src/topology/homotopy/pi_n.lean

```diff
@@ -66,220 +69,129 @@ The first projection of a non-zero dimensional cube
       congr, ext1, exact congr_fun hfg _,
     end }
 
-@[ext] lemma ext (f g : pre_π n x) (H : ∀ y, f y = g y) : f = g := fun_like.ext f g H
+@[ext] lemma ext (f g : nloop n x) (H : ∀ y, f y = g y) : f = g := fun_like.ext f g H
 
-@[simp] lemma mk_apply (f : C(I^n, X)) (H y) : (⟨f, H⟩ : pre_π n x)  y = f y := rfl
+@[simp] lemma mk_apply (f : C(I^n, X)) (H y) : (⟨f, H⟩ : nloop n x)  y = f y := rfl
 
 /--
-The constant `pre_π` at x
+The constant `nloop` at x
 -/
-def const : pre_π n x := ⟨⟨λ_x,by continuity⟩,λ_ _,rfl⟩
+def const : nloop n x := ⟨⟨λ_x,by continuity⟩,λ_ _,rfl⟩
 
-instance inhabited : inhabited (pre_π n x) :=
+instance inhabited : inhabited (nloop n x) :=
   { default := const }
 
 /--
-Homotopies are functions I×I^n → X
--/
-abbreviation homotopy (f g : pre_π n x) :=
-  f.to_continuous_map.homotopy_rel g.to_continuous_map (cube.boundary n)
-infix `~`:50 := homotopy
-
-namespace homotopy
-section
-variables  {f g h : pre_π n x} (H : f ~ g)
-
-instance fun_like : fun_like (f ~ g) (I^(n+1)) (λ _, X) :=
-  { coe := λ h c, h.1 (c 0, fin.tail c),
-    coe_injective' :=
-    begin
-      rintros h1 h2 hfg, cases h1, cases h2,
-      congr, ext1 ⟨x,y⟩, simpa only [fin.tail_cons] using congr_fun hfg (fin.cons x y),
-    end }
-
-@[continuity] lemma continuous : continuous H :=
-  by { refine H.continuous.comp _, continuity; apply continuous_apply }
-
-/--
-The constant homotopy at f
--/
-@[refl] def refl (f : pre_π n x) : f ~ f :=
-  continuous_map.homotopy_rel.refl _ _
-
-/--
-Reversing of a homotopy
--/
-@[symm] def symm : f ~ g → g ~ f :=
-  continuous_map.homotopy_rel.symm
-
-/--
-Concatenation of homotopies by using them along `[0,2⁻¹]` and `[2⁻¹,1] `
--/
-@[trans] def trans : f ~ g → g ~ h → f ~ h :=
-  continuous_map.homotopy_rel.trans
-
-/--
-Eval an homotopy at an intermediate point, yielding a `pre_π`
+Boundary-preserving homotopic relation between `nloop`s
 -/
-def eval (t:I) : pre_π n x :=
-  { to_fun := H.to_homotopy.curry t,
-    continuous_to_fun := by continuity,
-    boundary :=
-    begin
-      intros y H,
-      simp only [continuous_map.homotopy.curry_apply,
-        continuous_map.homotopy_with.coe_to_homotopy],
-      rw continuous_map.homotopy_rel.eq_fst,
-      apply f.boundary,
-      all_goals { assumption }
-    end }
-
-@[simp] lemma eval0 : H.eval 0 = f :=
-begin
-  ext, unfold eval,
-  simp only [mk_apply, continuous_map.homotopy_with.apply_zero,
-    continuous_map.coe_mk, continuous_map.homotopy.curry_apply,
-    continuous_map.homotopy_with.coe_to_homotopy],
-  reflexivity,
-end
-
-@[simp] lemma eval1 : H.eval 1 = g :=
-begin
-  ext, unfold eval,
-  simp only [mk_apply, continuous_map.coe_mk,
-    continuous_map.homotopy_with.apply_one, continuous_map.homotopy.curry_apply,
-    continuous_map.homotopy_with.coe_to_homotopy],
-  reflexivity,
-end
+def homotopic (f g : nloop n x) : Prop :=
+  nonempty (f.to_continuous_map.homotopy_rel g.to_continuous_map (cube.boundary n))
```

Comment:
Why not just `f.to_continuous_map.homotopic_rel g.to_continuous_map (cube.boundary n)`? It's defeq to this, and there are API for `homotopic_rel` parallel to `homotopy_rel`.